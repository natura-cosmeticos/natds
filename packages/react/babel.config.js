const path = require('path')

const defaultAlias = {
  '@naturacosmeticos/natds-react': path.resolve(__dirname, './src/components')
}

module.exports = {
  plugins: [
    ['babel-plugin-module-resolver', { alias: defaultAlias }]
  ],
  presets: [
    ['@babel/preset-env', { targets: { node: 'current' } }],
    '@babel/preset-typescript',
    '@babel/preset-react'
  ]
}
