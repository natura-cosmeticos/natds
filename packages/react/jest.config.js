const base = require('../../jest.config.base.js')
const { displayName, name } = require('./package.json')

module.exports = {
  ...base,
  collectCoverageFrom: [
    'src/**/*.{js,jsx,ts,tsx}',
    '!src/**/*.stories.tsx',
    '!src/helpers/**'
  ],
  errorOnDeprecated: true,
  displayName,
  name,
  roots: ['<rootDir>/src'],
  transform: {
    '^.+\\.[tj]sx?$': 'babel-jest',
    '^.+\\.mdx?$': '@storybook/addon-docs/jest-transform-mdx'
  }
}
