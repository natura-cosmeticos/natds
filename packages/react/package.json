{
  "name": "@naturacosmeticos/natds-react",
  "displayName": "NatDS-React",
  "description": "A collection of components from Natura Design System for React",
  "version": "1.8.0-alpha.DSY-1959.4.0",
  "private": false,
  "keywords": [
    "design-system",
    "natura",
    "react",
    "web"
  ],
  "homepage": "https://github.com/natura-cosmeticos/natds-js/tree/master/packages/react/README.md",
  "bugs": {
    "url": "https://github.com/natura-cosmeticos/natds-js/issues"
  },
  "license": "ISC",
  "author": {
    "email": "designsystem@natura.net",
    "name": "Natura&Co.",
    "url": "https://github.com/natura-cosmeticos"
  },
  "files": [
    "README.md",
    "dist/"
  ],
  "jsdelivr": "dist/umd/index.js",
  "source": "src/index.ts",
  "types": "dist/index.d.ts",
  "typings": "dist/index.d.ts",
  "umd:main": "dist/umd/index.js",
  "unpkg": "dist/umd/index.js",
  "main": "dist/index.js",
  "directories": {
    "lib": "dist"
  },
  "repository": {
    "directory": "packages/react",
    "type": "git",
    "url": "git+https://github.com/natura-cosmeticos/natds-js.git"
  },
  "scripts": {
    "build": "yarn clean && tsc -p ./tsconfig.json && yarn rollup:build",
    "build:storybook": "yarn build-storybook -o docs/dist",
    "build:storybook:ci": "yarn build:storybook --quiet",
    "clean": "rm -rf ./dist",
    "copy:assets": "mkdir -p ./dist/assets && cp -r ./src/assets/* ./dist/assets",
    "lint": "eslint --ext .js,.jsx,.ts,.tsx ./src --color",
    "lint:fix": "yarn lint --fix",
    "rollup:build": "rollup -c",
    "storybook": "start-storybook -p 6006",
    "test": "jest --verbose",
    "test:ci": "jest --coverage && codecov -t",
    "test:watch": "jest --watch"
  },
  "dependencies": {
    "@naturacosmeticos/natds-icons": "1.3.0",
    "@naturacosmeticos/natds-themes": "^0.4.12",
    "react-jss": "10.5.1"
  },
  "devDependencies": {
    "@types/react": "17.0.3",
    "@types/react-dom": "17.0.3",
    "react": "17.0.2",
    "react-dom": "17.0.2",
    "react-test-renderer": "17.0.2"
  },
  "peerDependencies": {
    "@types/react": ">= 16.8.6",
    "@types/react-dom": "17.0.3",
    "react": ">= 16.8.4",
    "react-dom": ">= 16.8.4"
  },
  "peerDependenciesMeta": {
    "@types/react": {
      "optional": true
    },
    "@types/react-dom": {
      "optional": true
    }
  }
}
