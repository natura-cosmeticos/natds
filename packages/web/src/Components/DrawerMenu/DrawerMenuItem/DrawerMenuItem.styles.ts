import { createStyles, makeStyles } from '@material-ui/core'
import { IThemeWeb } from '../../../Themes'

const style = (theme: IThemeWeb) => createStyles({
  listItem: {
    borderRadius: theme.shape?.borderRadius,
    cursor: 'pointer',
    padding: theme.sizes?.tiny,
    '&:hover': {
      backgroundColor: theme.palette?.action?.hover
    }
  },
  iconItem: {
    marginRight: theme.sizes?.standard,
    marginTop: theme.sizes?.none,
    minWidth: 'auto'
  },
  subheader: {
    backgroundColor: `${theme.palette?.background?.paper}`,
    borderTop: `${theme.palette?.action?.hover} 1px solid`,
    margin: `${theme.sizes?.small}px -${theme.sizes?.tiny}px 0`
  },
  linkItem: {
    width: '100%',
    textDecoration: 'none',
    color: 'inherit'
  }
})

export const useStyles = makeStyles(style, { name: 'NatDSMenuItemText' })

export default useStyles
