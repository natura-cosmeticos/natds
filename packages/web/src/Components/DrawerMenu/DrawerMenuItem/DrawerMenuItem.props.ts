import * as React from 'react'
import { IconName } from '@naturacosmeticos/natds-icons'
import { IThemeWeb } from '../../../Themes'

export interface IDrawerMenuItemProps {
  badge?: string;
  icon?: IconName;
  list?: Array<unknown>;
  name?: string;
  onSelect?: (event: React.SyntheticEvent, name?: string) => void;
  onToggle?: (event: React.SyntheticEvent, state: boolean, name?: string) => void;
  opened?: boolean;
  section?: string;
  selected?: boolean;
  href?: string

  /**
   * @deprecated `theme` property is deprecated since 0.22
   *
   * Please wrap your application with `<Provider>` component.
   */
  theme?: IThemeWeb | unknown
}
