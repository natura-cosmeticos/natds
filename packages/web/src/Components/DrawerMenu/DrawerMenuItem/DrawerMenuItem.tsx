/* eslint-disable @typescript-eslint/ban-ts-comment */
/* eslint-disable complexity */
import * as React from 'react'
import ListSubheader from '../../ListSubheader'
import Typography from '../../Typography'
import ListItem from '../../ListItem'
import ListItemIcon from '../../ListItemIcon'
import { Icon } from '../../Icon'
import ListItemText from '../../ListItemText'
import { IDrawerMenuItemProps } from './DrawerMenuItem.props'
import { MenuItemText } from '../MenuItemText'
import { IDrawerMenuSectionProps } from '../DrawerMenuSection'
import useStyles from './DrawerMenuItem.styles'

export const DrawerMenuItem = React.forwardRef<HTMLDivElement, IDrawerMenuItemProps>((
  props: IDrawerMenuItemProps, ref
) => {
  const {
    onSelect,
    name,
    icon,
    selected,
    section,
    href,
    onClickLink
  } = props

  const {
    listItem, iconItem, subheader, linkItem
  } = useStyles()

  if (section) {
    return (
      <ListSubheader className={subheader}>
        <Typography variant="caption" component="span">{section}</Typography>
      </ListSubheader>
    )
  }

  const handleClick = (event: React.SyntheticEvent) => onSelect && onSelect(event, name)
  const onClick = onSelect ? { onClick: handleClick } : {}

  return (
    <ListItem
      {...onClick}
      className={listItem}
      selected={selected}
      // @ts-ignore
      component="li"
      button
      ref={ref}
    >
      {href
        ? (
          <a href={href} className={linkItem} onClick={onClickLink}>
            {icon && <ListItemIcon className={iconItem}><Icon name={icon} size="tiny" /></ListItemIcon>}
            {name}
          </a>
        )
        : (
          <>
            {icon && <ListItemIcon className={iconItem}><Icon name={icon} size="tiny" /></ListItemIcon>}
            <ListItemText>
              <MenuItemText {...props as IDrawerMenuSectionProps} />
            </ListItemText>

          </>
        )}
    </ListItem>
  )
})

DrawerMenuItem.displayName = 'DrawerMenuItem'

export default DrawerMenuItem
