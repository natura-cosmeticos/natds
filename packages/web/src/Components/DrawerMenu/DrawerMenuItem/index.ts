import DrawerMenuItem from './DrawerMenuItem'

export { IDrawerMenuItemProps } from './DrawerMenuItem.props'
export { DrawerMenuItem }
export default DrawerMenuItem
