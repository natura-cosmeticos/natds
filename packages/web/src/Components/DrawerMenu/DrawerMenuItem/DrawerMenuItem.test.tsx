import * as React from 'react'
import * as TestRenderer from 'react-test-renderer'
import DrawerMenuItem from './DrawerMenuItem'

describe('Drawer Menu Item component', () => {
  it('should match snapshot', () => {
    const testRenderer : TestRenderer.ReactTestRenderer = TestRenderer.create(<DrawerMenuItem />)

    expect(testRenderer).toMatchSnapshot()
  })
})
