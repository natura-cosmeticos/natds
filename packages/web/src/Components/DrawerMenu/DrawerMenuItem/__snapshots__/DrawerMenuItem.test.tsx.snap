// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer Menu Item component should match snapshot 1`] = `
<li
  aria-disabled={false}
  className="MuiButtonBase-root MuiListItem-root NatDSMenuItemText-listItem-1 MuiListItem-gutters MuiListItem-button"
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="button"
  tabIndex={0}
>
  <div
    className="MuiListItemText-root"
  >
    <span
      className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
    >
      <span
        className="MuiTypography-root MuiTypography-body2"
      />
    </span>
  </div>
</li>
`;
